jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/UploadToLambdaModal.kt

```diff
@@ -127,7 +129,15 @@ class UploadToLambdaController(
                 .map { IamRole(name = it.roleName(), arn = it.arn()) }
                 .toList()
         }
-        view.sourceBucket.populateValues { s3Client.listBuckets().buckets().filterNotNull().mapNotNull { it.name() } }
+        view.sourceBucket.populateValues {
+            val activeRegionId = ProjectAccountSettingsManager.getInstance(project).activeRegion.id
+            s3Client.listBuckets().buckets()
+                .asSequence()
+                .filterNotNull()
+                .mapNotNull { it.name() }
+                .filter { s3Client.regionForBucket(it) == activeRegionId }
```

Comment:
Would it make sense to unconditionally include the global buckets?

Reply:
Unsure if that works or not...

Reply:
Ok, I might have been thinking of an outdated concept